python/lsst/ctrl/bps/htcondor/htcondor_service.py

```diff
@@ -1673,6 +1682,55 @@ def _wms_id_to_cluster(wms_id):
     return schedd_ad, cluster_id, id_type
 
 
+def _wms_id_to_dir(wms_id):
+    """Convert WMS id to submit directory.
+
+    Parameters
+    ----------
+    wms_id : `int` or `float` or `str`
+        HTCondor job id or path.
+
+    Returns
+    -------
+    wms_dir : `str`
+        Submit directory for the run with the given job id. If the submit
+        directory cannot be determined, it will be set to None.
```

Comment:
The submit directory can't be determined if the user provided a local job id as the run id, but there are no entries for this run in the HTCondor history file either because it rolled over or we're querying a wrong scheduler because, for example, the user ran `bps restart` on `sdfrome02` while the workflow was initially submitted to execution on `sdfrome01`.